framework/src/modules/token/api.ts

```diff
@@ -52,55 +63,87 @@ export class TokenAPI extends BaseAPI {
 		apiContext: APIContext,
 		senderAddress: Buffer,
 		recipientAddress: Buffer,
-		_tokenID: TokenID,
+		tokenID: TokenID,
 		amount: bigint,
 	): Promise<void> {
+		const canonicalTokenID = await this._getCanonicalTokenID(apiContext, tokenID);
 		const userStore = apiContext.getStore(this.moduleID, STORE_PREFIX_USER);
-		const sender = await userStore.getWithSchema<UserStoreData>(senderAddress, userStoreSchema);
-		if (sender.availableBalance < amount + this._minBalance) {
+		const sender = await userStore.getWithSchema<UserStoreData>(
+			getUserStoreKey(senderAddress, canonicalTokenID),
+			userStoreSchema,
+		);
+		if (sender.availableBalance < amount) {
 			throw new Error(
 				`Sender ${senderAddress.toString(
 					'hex',
-				)} balance ${sender.availableBalance.toString()} is not sufficient for ${(
-					amount + this._minBalance
-				).toString()}`,
+				)} balance ${sender.availableBalance.toString()} is not sufficient for ${amount.toString()}`,
 			);
 		}
+
 		sender.availableBalance -= amount;
-		await userStore.setWithSchema(senderAddress, sender, userStoreSchema);
+
+		await userStore.setWithSchema(
+			getUserStoreKey(senderAddress, canonicalTokenID),
+			sender,
+			userStoreSchema,
+		);
 
 		let recipient: UserStoreData;
+		let recipientExist = false;
 		try {
-			recipient = await userStore.getWithSchema<UserStoreData>(recipientAddress, userStoreSchema);
+			recipient = await userStore.getWithSchema<UserStoreData>(
+				getUserStoreKey(recipientAddress, canonicalTokenID),
+				userStoreSchema,
+			);
+			recipientExist = true;
 		} catch (error) {
 			if (!(error instanceof NotFoundError)) {
 				throw error;
 			}
+
 			recipient = {
 				availableBalance: BigInt(0),
 				lockedBalances: [],
 			};
 		}
-		if (recipient.availableBalance + amount < this._minBalance) {
-			throw new Error(
-				`Recipient ${recipientAddress.toString('hex')} balance ${(
-					recipient.availableBalance + amount
-				).toString()} is not sufficient for min balance ${this._minBalance.toString()}`,
-			);
+		const minBalance = this._getMinBalance(tokenID);
```

Comment:
If i understand the function correctly, this would set minBalance =0 if the tokenID is not present in the minBalances config.
Here we would rather want the transfer to fail if `!recipientExist` as this transfer is not able to pay to initialize the account.